### DISCUSSION TITLE
Scalable Keyboard

### DISCUSSION THREAD

**Discussion by aetkie** [UNC-Project-Open-AAC/OS-DPI#250] [Show and tell]
**This is a UI idea that our team has been exploring based on a conversation with a member of the Project Open community with valuable experience as an expert AAC user.**

Currently, he switches between 2 direct selection methods depending on his position - (1) eye gaze and (2) a trackball mouse. 

For this individual, moving from sitting to lying presents with the challenge of not being able to auto-rotate the onscreen keyboard to accommodate a change in position (_think, moving your phone from portrait to landscape mode to take advantage of real estate_). 

We thought it would be interesting to explore a keyboard that can scale up and down in size to accommodate access and position changes. Instead of taking advantage of real estate, we worked on _condensing_ the keyboard so that he could make micro movements and access the keys.

The scalable keyboard UI went through many iterations and this is the most current:

[autoscalable_kybd_050624_AE.zip](https://github.com/UNC-Project-Open-AAC/OS-DPI/files/15227891/autoscalable_kybd_050624_AE.zip)

**Comment by VanSickleh1** [comment]
Hey @aetkie! What a wonderful design solution. I am a quadriplegic and use eye gaze software as well as the member you mentioned. I am far from a computer with, but tend to know just enough to be dangerous or to really screw something up. 😁 I just finished law school and in the process of studying for the bar, so I am on my computer almost 24/7. A scalable keyboard would be a very useful asset to me as well. I downloaded your zip file. Can you please point me in the right direction of _read me_ style tutorial on how to install the keyboard and begin using it? And while I'm at it, have you seen any other great design solutions for people with severe physical disabilities to maximize their effectiveness on their computer? I have a million ideas and maybe one or two of them have already been solved. 🤞 Thank you so much for Designing this and I look forward to hearing from you!

**Reply by sofia-bg** [reply]
Hi @VanSickleh1 I made a few updates to the version that @gbishop shared below. I noticed that on our "change access" button we only had the size of the keyboard change, not the access method. Also the clear and delete buttons weren't working on the small version of the key board. 

We have it set up now so that when they keyboard is large you can use the front facing camera to control the mouse, our emulation of using eye tracking on the larger keypad. Right now we only have the keyboard set up as an AAC system, so after you type you can click on the message window to have what was written speak (or the speak button in the lower right hand corner works for this purpose as well). We do not currently have the keyboard set up to type into other programs, which I think is what you were asking to be able to do? 

[scalable keyboard](https://unc-project-open-aac.github.io/OS-DPI/?fetch=https://github.com/user-attachments/files/15873800/autoscalable_kybd_061724-1.zip)

**Reply by aetkie** [reply]
@VanSickleh1 Looks like my teammates stepped in to answer your questions about the scalable kb. 

As for your millions of ideas about great design solutions, if you'd like, you are welcome to share a few in [a new discussion post](https://github.com/UNC-Project-Open-AAC/OS-DPI/discussions/categories/ideas). Maybe a designer in this Project Open Community has either designed a solution or someone else could suggest a solution. Thanks for checking out our design!

**Comment by gbishop** [comment]
You should be able to try out that design using this [link](https://unc-project-open-aac.github.io/OS-DPI/?fetch=https://github.com/UNC-Project-Open-AAC/OS-DPI/files/15227891/autoscalable_kybd_050624_AE.zip)
